### PR TITLE
Allow DroneCore to be added to built by a parent cmake project

### PR DESCRIPTION
**Pull request by Stifael**
This allows DroneCore to be added to any cmake project just by doing:
```
add_subdirectory(path/to/dronecore)
```

### CODE REVIEW

File: core/CMakeLists.txt [hunk 1]
```diff
@@ -83,17 +83,17 @@ install(FILES
...
-    ${CMAKE_SOURCE_DIR}/core/call_every_handler_test.cpp
-    ${CMAKE_SOURCE_DIR}/core/curl_test.cpp
-    ${CMAKE_SOURCE_DIR}/core/any_test.cpp
-    ${CMAKE_SOURCE_DIR}/core/cli_arg_test.cpp
-    ${CMAKE_SOURCE_DIR}/core/locked_queue_test.cpp
-    ${CMAKE_SOURCE_DIR}/core/thread_pool_test.cpp
+    ${PROJECT_SOURCE_DIR}/core/global_include_test.cpp
+    ${PROJECT_SOURCE_DIR}/core/mavlink_channels_test.cpp
+    ${PROJECT_SOURCE_DIR}/core/unittests_main.cpp
+    # TODO: add this again
```
Comment (julianoes) [inline core/CMakeLists.txt]: Please delete the trailing white space.

File: plugins/mission/CMakeLists.txt [hunk 2]
```diff
@@ -6,8 +6,8 @@ add_library(dronecore_mission ${PLUGIN_LIBRARY_TYPE}
 )
 
 include_directories(
-    ${CMAKE_SOURCE_DIR}/core
-    SYSTEM ${CMAKE_SOURCE_DIR}/third_party/include
+    #${CMAKE_CURRENT_SOURCE_DIR}/core
+    #SYSTEM ${CMAKE_CURRENT_SOURCE_DIR}/third_party/include
```
Comment (julianoes) [inline plugins/mission/CMakeLists.txt]: Fix the accidental `#` changes here.

**Comment by JonasVautherin** [comment]
Thanks! Why is it a [WIP]? Can't we merge it?

**Comment by Stifael** [comment]
> Why is it a [WIP]? Can't we merge it?

Additional missing replacements. Should be ok now

**Comment by julianoes** [comment]
@JonasVautherin oh looks like Jenkins forgot how to do clang-format.

**Comment by julianoes** [comment]
Waiting with merging until we figure the Jenkinsbuild out in #420.

**Comment by julianoes** [comment]
@Stifael please rebase, CI should be fixed.

**Pull request merged**